demo/util/fps-display.mjs

```diff
@@ -0,0 +1,51 @@
+const TEMPLATE = `
+<style>
+  .fps { width: 23%; float: left; text-align: right; padding-right: 2% }
+</style>
+
+  <div id=fps>
+    <div class=fps><span class=place>0</span> FPS</div>
+    <div class=fps><span class=place>0</span> FPS</div>
+    <div class=fps><span class=place>0</span> FPS</div>
+    <div class=fps><span class=place>0</span> FPS</div>
+  </div>
+`;
+
+const MS_PER_SECOND = 1000;
+
+/**
+ * Displays frames-per-second performance of the page for recent frames.
+ */
+export class FpsDisplay extends HTMLElement {
+  #lastTime = null;
+  #counter = 0;
+  #fps
+
+  constructor() {
+    super();
+
+    const shadowRoot = this.attachShadow({mode: 'closed'});
+    shadowRoot.innerHTML = TEMPLATE;
+    this.#fps = shadowRoot.getElementById('fps')
+      .getElementsByClassName('place');
+    this.#scheduleUpdate();
```

Comment:
It seems you could use everyFrame (and remove scheduleUpdate) if you modified everyFrame to pass in the timestamp as well as the index. That might also remove #counter.

Reply:
That's much nicer.